pkg/services/datasources/service/datasource.go

```diff
@@ -236,6 +249,28 @@ func (s *Service) AddDataSource(ctx context.Context, cmd *datasources.AddDataSou
 	})
 }
 
+// getAvailableName finds the first available name for a datasource of the given type.
+func getAvailableName(dsType string, dataSources []*datasources.DataSource) string {
+	dsNames := make(map[string]bool)
+	for _, ds := range dataSources {
+		dsNames[strings.ToLower(ds.Name)] = true
+	}
+
+	name := dsType
+	currentDigit := 0
+
+	for dsNames[strings.ToLower(name)] {
+		if currentDigit%10 == 0 {
+			currentDigit = 1
+			name = name + "-x"
+		}
+		name = fmt.Sprintf("%s%d", name[:len(name)-1], currentDigit)
+		currentDigit++
+	}
```

Comment:
Currently the code would produces names like `prometheus-9-9-9-3`. This is sticking to how the old logic worked. Alternatively, we could change it to just keep incrementing the number (eg, we could have `prometheus-1234`). It seems simpler to me, but I don't know what the original reasoning was.

Reply:
I guess that it is/was compared in alphabetical order?

Reply:
Good question! I just realised that the old numbering would not have been in alphabetical order (`prometheus-8` is alphabetically after `prometheus-10-1`, but would have been created before it). The new numbering (counting up to 9 before hyphenating again) would result in correct alphabetical order.

That said, the more I think about it, the better it seems to just keep incrementing the number without adding more hyphens (so `prometheus-9`, `prometheus-10`, `prometheus-11` etc). I'll rewrite the code to do that. Let me know if you or @leventebalogh (the author of the original code) have any objections :)